core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -601,29 +620,30 @@ private PointInTimeRecoverySpecification buildPointInTimeRecoverySpecification()
   }
 
   private UpdateContinuousBackupsRequest buildUpdateContinuousBackupsRequest(
-      String namespace, String table) {
+      Namespace namespace, String table) {
     return UpdateContinuousBackupsRequest.builder()
         .tableName(getFullTableName(namespace, table))
         .pointInTimeRecoverySpecification(buildPointInTimeRecoverySpecification())
         .build();
   }
 
   @Override
-  public void dropTable(String namespace, String table) throws ExecutionException {
-    disableAutoScaling(namespace, table);
+  public void dropTable(String nonPrefixedNamespace, String table) throws ExecutionException {
+    Namespace prefixedNamespace = Namespace.of(namespacePrefix, nonPrefixedNamespace);
+    disableAutoScaling(prefixedNamespace, table);
 
-    String fullTableName = getFullTableName(namespace, table);
+    String fullTableName = getFullTableName(prefixedNamespace, table);
     try {
       client.deleteTable(DeleteTableRequest.builder().tableName(fullTableName).build());
     } catch (Exception e) {
       throw new ExecutionException("deleting table " + fullTableName + " failed", e);
     }
-    waitForTableDeletion(namespace, table);
-    deleteTableMetadata(namespace, table);
+    waitForTableDeletion(prefixedNamespace, table);
+    deleteTableMetadata(prefixedNamespace, table);
   }
 
-  private void disableAutoScaling(String namespace, String table) throws ExecutionException {
-    TableMetadata tableMetadata = getTableMetadata(namespace, table);
+  private void disableAutoScaling(Namespace namespace, String table) throws ExecutionException {
+    TableMetadata tableMetadata = getTableMetadata(namespace.nonPrefixed(), table);
```

Comment:
The main reason for adding the `Namespace` type was that private methods end up calling interface methods that are all requiring a nonprefixed namespace. This limits the possibility of introducing bugs by passing a prefixed namespace to the interface methods.